src/commons/sagas/PersistenceSaga.tsx

```diff
@@ -307,35 +325,101 @@ const initialisationPromise: Promise<void> = new Promise(res => {
   startInitialisation = res;
 }).then(initialise);
 
-function handleUserChanged(user: gapi.auth2.GoogleUser) {
-  store.dispatch(
-    actions.setGoogleUser(user.isSignedIn() ? user.getBasicProfile().getEmail() : undefined)
-  );
-}
-
+// only called once
 async function initialise() {
-  await new Promise((resolve, reject) =>
-    gapi.load('client:auth2', { callback: resolve, onerror: reject })
+  // load GIS script
+  // adapted from https://github.com/MomenSherif/react-oauth
+  await new Promise<void>((resolve, reject) => {
+    const scriptTag = document.createElement('script');
+    scriptTag.src = 'https://accounts.google.com/gsi/client';
+    scriptTag.async = true;
+    scriptTag.defer = true;
+    scriptTag.onload = () => resolve();
+    scriptTag.onerror = ev => {
+      reject(ev);
+    };
+    document.body.appendChild(scriptTag);
+  });
```

Comment:
We should follow React declarative-style coding conventions, instead of plain JS conventions. We can just use JSX to render the script tag directly.